### PR TITLE
python3Packages.spacy_models.*: add Croatian, Korean, Finnish and Ukrainian language models

### PR DESCRIPTION
**Pull request by jboynyc**
###### Description of changes

Update spaCy [language models](https://spacy.io/models/) to versions 3.5.0 and add language models for Croatian, Korean, Finnish, and Ukrainian that have been added upstream. 

Currently nixpkgs contains version 3.3 language models which don't work with spaCy 3.5 (which nixpkgs already has).

As in the [previous update](https://github.com/NixOS/nixpkgs/pull/91217) by @danieldk, I am omitting the Japanese models because of the difficulty of packaging `sudachidict-core` and `sudachipy`.

A new dependency that I did include is `pymorphy3`, required by the Russian and Ukrainian models. nixpkgs already contained `pymorphy2`, of which `pymorphy3` is a fork, and the derivation needed for pymorphy3 is virtually unchanged.

I also updated the `spacy-transformers` package because the transformer models require a newer release than the one currently in nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jboynyc** [comment]
Result of nixpkgs-review run on x86_64-linux 1
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python310Packages.pymorphy3</li>
    <li>python310Packages.pymorphy3-dicts-ru</li>
    <li>python310Packages.pymorphy3-dicts-uk</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python311Packages.pymorphy3</li>
    <li>python311Packages.pymorphy3-dicts-ru</li>
    <li>python311Packages.pymorphy3-dicts-uk</li>
  </ul>
</details>

**Comment by jboynyc** [comment]
[Here](https://gist.github.com/jboynyc/8576d2ad86eb6af415dd9e1fa468fcc8) is a script to update the `models.json` file.

**Comment by jboynyc** [comment]
The full list of language models after this update is:

* ca_core_news_lg
* ca_core_news_md
* ca_core_news_sm
* ca_core_news_trf
* da_core_news_lg
* da_core_news_md
* da_core_news_sm
* da_core_news_trf
* de_core_news_lg
* de_core_news_md
* de_core_news_sm
* de_dep_news_trf
* el_core_news_lg
* el_core_news_md
* el_core_news_sm
* en_core_web_lg
* en_core_web_md
* en_core_web_sm
* en_core_web_trf
* es_core_news_lg
* es_core_news_md
* es_core_news_sm
* es_dep_news_trf
* fi_core_news_lg
* fi_core_news_md
* fi_core_news_sm
* fr_core_news_lg
* fr_core_news_md
* fr_core_news_sm
* fr_dep_news_trf
* hr_core_news_lg
* hr_core_news_md
* hr_core_news_sm
* it_core_news_lg
* it_core_news_md
* it_core_news_sm
* ko_core_news_lg
* ko_core_news_md
* ko_core_news_sm
* lt_core_news_lg
* lt_core_news_md
* lt_core_news_sm
* mk_core_news_lg
* mk_core_news_md
* mk_core_news_sm
* nb_core_news_lg
* nb_core_news_md
* nb_core_news_sm
* nl_core_news_lg
* nl_core_news_md
* nl_core_news_sm
* pl_core_news_lg
* pl_core_news_md
* pl_core_news_sm
* pt_core_news_lg
* pt_core_news_md
* pt_core_news_sm
* ro_core_news_lg
* ro_core_news_md
* ro_core_news_sm
* ru_core_news_lg
* ru_core_news_md
* ru_core_news_sm
* sv_core_news_lg
* sv_core_news_md
* sv_core_news_sm
* uk_core_news_lg
* uk_core_news_md
* uk_core_news_sm
* uk_core_news_trf
* xx_ent_wiki_sm
* xx_sent_ud_sm                                                                                                                                               
* zh_core_web_lg                                                                                                                                              
* zh_core_web_md                                                                                                                                              
* zh_core_web_sm                                                                                                                                              
* zh_core_web_trf

File: pkgs/top-level/python-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:10769]:
```suggestion
  spacy_models = callPackage ../development/python-modules/spacy/models.nix { };
```

File: pkgs/development/python-modules/pymorphy3/dicts-uk.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymorphy3/dicts-uk.nix:21]: We cannot add packages without a maintainer

File: pkgs/development/python-modules/pymorphy3/dicts-ru.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymorphy3/dicts-ru.nix:21]: we cannot add packages without a maintainer

File: pkgs/development/python-modules/pymorphy3/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymorphy3/default.nix:35]: and same here

File: pkgs/development/python-modules/pymorphy3/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymorphy3/default.nix:15]: We no longer support python2. Please use pythonOlder/pythonAtLeast and a version number

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pymorphy3/default.nix
Comment (jboynyc) [reply pkgs/development/python-modules/pymorphy3/default.nix]: Good catch. Looks like all versions of python3 are supported, so I removed `disabled` altogether.

**Comment by jboynyc** [comment]
rebased after merge of #186633

**Review by jboynyc — COMMENTED** [review]
all changes made as suggested

**Comment by SuperSandro2000** [comment]
I removed the merge commit because that can cause really hairy issues when rebasing/merging other branches. I hope I didn't do a mistake.

**Comment by jboynyc** [comment]
My bad, I did a merge instead of a rebase. I'll try again.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.spacy-transformers

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 215176` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python310Packages.spacy-transformers: log was empty</li>
  </ul>
</details>
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python310Packages.pymorphy3</li>
<li>python310Packages.pymorphy3-dicts-ru</li>
<li>python310Packages.pymorphy3-dicts-uk</li>
<li>python311Packages.pymorphy3</li>
<li>python311Packages.pymorphy3-dicts-ru</li>
<li>python311Packages.pymorphy3-dicts-uk</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python310Packages.pymorphy3-dicts-ru:
<details>

got build log for '/nix/store/a07sbniranwmplgcmylp8h1k59zxjjhk-python3.10-pymorphy3-dicts-ru-2.4.417150.4580142' from 'daemon'
When evaluating attribute ‘python310Packages.pymorphy3-dicts-ru’:
warning: no-python-tests
Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pymorphy3/dicts-ru.nix:18:

```
   |
18 |     description = "Russian dictionaries for pymorphy3";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md
</details>
python310Packages.pymorphy3-dicts-uk:
<details>

got build log for '/nix/store/maj3hn9x2499rv1y5i1arghb7z2yc9zy-python3.10-pymorphy3-dicts-uk-2.4.1.1.1663094765' from 'daemon'
When evaluating attribute ‘python310Packages.pymorphy3-dicts-uk’:
warning: no-python-tests
Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pymorphy3/dicts-uk.nix:18:

```
   |
18 |     description = "Ukrainian dictionaries for pymorphy3";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md
</details>
python310Packages.pymorphy3:
<details>

got build log for '/nix/store/61xh5q4vrg0rg0kdcvmknk5irj562qa7-python3.10-pymorphy3-1.2.0' from 'daemon'
When evaluating attribute ‘python310Packages.pymorphy3’:
warning: no-python-tests
Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pymorphy3/default.nix:29:

```
   |
29 |     description = "Morphological analyzer/inflection engine for Russian and Ukrainian";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md
</details>
python311Packages.pymorphy3-dicts-ru:
<details>

got build log for '/nix/store/khfh4vndpp8j1pdm5yvbv70a4g3vflnb-python3.11-pymorphy3-dicts-ru-2.4.417150.4580142' from 'daemon'
When evaluating attribute ‘python311Packages.pymorphy3-dicts-ru’:
warning: no-python-tests
Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pymorphy3/dicts-ru.nix:18:

```
   |
18 |     description = "Russian dictionaries for pymorphy3";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md
</details>
python311Packages.pymorphy3-dicts-uk:
<details>

got build log for '/nix/store/dcva9ki54wzqz5flad2lw27ydi3an62z-python3.11-pymorphy3-dicts-uk-2.4.1.1.1663094765' from 'daemon'
When evaluating attribute ‘python311Packages.pymorphy3-dicts-uk’:
warning: no-python-tests
Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pymorphy3/dicts-uk.nix:18:

```
   |
18 |     description = "Ukrainian dictionaries for pymorphy3";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md
</details>
python311Packages.pymorphy3:
<details>

got build log for '/nix/store/az1b8rmp1qfkw7iwqvx5b0wqkyan1by1-python3.11-pymorphy3-1.2.0' from 'daemon'
When evaluating attribute ‘python311Packages.pymorphy3’:
warning: no-python-tests
Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pymorphy3/default.nix:29:

```
   |
29 |     description = "Morphological analyzer/inflection engine for Russian and Ukrainian";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md
</details>
</details>

**Comment by jboynyc** [comment]
* Switch `pymorphy3` from `fetchPypi` to `fetchFromGitHub` and run tests with pytest hook. 
* Set `doCheck = false` for the pymorphy3-dict packages, which don't have any tests.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python310Packages.pymorphy3</li>
    <li>python310Packages.pymorphy3-dicts-ru</li>
    <li>python310Packages.pymorphy3-dicts-uk</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python311Packages.pymorphy3</li>
    <li>python311Packages.pymorphy3-dicts-ru</li>
    <li>python311Packages.pymorphy3-dicts-uk</li>
  </ul>
</details>

**Comment by wegank** [comment]
@ofborg build python310Packages.spacy-transformers

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 215176` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python310Packages.pymorphy3</li>
    <li>python310Packages.pymorphy3-dicts-ru</li>
    <li>python310Packages.pymorphy3-dicts-uk</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python311Packages.pymorphy3</li>
    <li>python311Packages.pymorphy3-dicts-ru</li>
    <li>python311Packages.pymorphy3-dicts-uk</li>
  </ul>
</details>

File: pkgs/development/python-modules/spacy/models.nix
Comment (wegank) [inline pkgs/development/python-modules/spacy/models.nix:39]:
```suggestion
      postPatch = lib.optionalString requires-protobuf ''
```

File: pkgs/top-level/python-packages.nix
Comment (wegank) [inline pkgs/top-level/python-packages.nix:8434]: Is the package unused?

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 215176` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python310Packages.pymorphy3</li>
    <li>python310Packages.pymorphy3-dicts-ru</li>
    <li>python310Packages.pymorphy3-dicts-uk</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python311Packages.pymorphy3</li>
    <li>python311Packages.pymorphy3-dicts-ru</li>
    <li>python311Packages.pymorphy3-dicts-uk</li>
  </ul>
</details>

**Comment by wegank** [comment]
@ofborg build python310Packages.spacy-transformers.passthru.tests

**Comment by wegank** [comment]
`test_verbs` needs to be updated:


```python
def test_verbs(doc_en_core_web_trf):
    assert [
        token.lemma_ for token in doc_en_core_web_trf if token.pos_ == "VERB"] == [
        'start',
        'work',
        'drive',
        'take',
        'tell',
        'shake',
        'turn',
        'talk',
        'say']
```

File: pkgs/development/python-modules/pymorphy3/dicts-uk.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymorphy3/dicts-uk.nix:15]:
```suggestion
  # has no tests
  doCheck = false;
```

File: pkgs/development/python-modules/pymorphy3/dicts-ru.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymorphy3/dicts-ru.nix:15]:
```suggestion
  # has no tests
  doCheck = false;
```

**Comment by SuperSandro2000** [comment]
> `test_verbs` needs to be updated:

That will hopefully be the last round of reviews.

**Pull request merged**
